### DISCUSSION TITLE
Interactive ripgrep guide and playground

### DISCUSSION THREAD

**Discussion by nalgeon** [BurntSushi/ripgrep#2758] [General] [answered]
Hi Andrew! Thank you for this great project!

I've made an interactive ripgrep tutorial, so newcomers can try it right from the browser without installing anything: [Try ripgrep in Y minutes](https://codapi.org/try/ripgrep/) ([source](https://github.com/nalgeon/tryxinyminutes/blob/main/try/ripgrep/index.md)).

I've also created a navigable ripgrep [playground](https://codapi.org/ripgrep/).

I hope the tutorial and the playground will be useful for new ripgrep users. What do you think?

**Comment by BurntSushi** [comment, marked as answer]
Nice, very cool. The interactive examples are nice.

Is there a way to make it show ripgrep's default output format? Colors might be difficult to capture though unfortunately.

**Reply by nalgeon** [reply]
Thank you! I think the colors are possible, but they will require an additional conversion step (ansi → html). I'll look into it.

**Reply by nalgeon** [reply]
Done! Added the `--pretty` option behind the scenes to show line numbers, headings and colors by default.

<img width="400" alt="ripgrep" src="https://github.com/BurntSushi/ripgrep/assets/2946182/04ac3236-e25a-4534-add9-691a48dc9041">

If you think this is useful for ripgrep newbies, would you consider adding a link to the guide and/or playground in the readme? It's totally cool if you don't, of course.

**Reply by BurntSushi** [reply]
Yeah I think that sounds good. I am a little concerned about it disappearing since it's so new, but I don't think that should stop us from adding it. How about adding a new section with a link to it just before `Installation` and just after [`Feature comparison`](https://github.com/BurntSushi/ripgrep?tab=readme-ov-file#feature-comparison)?

**Reply by nalgeon** [reply]
Thanks a lot! I've been building interactive guides since September 2023, which is definitely not a very long time :) I have no plans to abandon this initiative — but if I ever do, I will give you a heads up so you can remove the link (or I can always help you self-host the sandbox if you wish).

A section between "Feature comparison" and "Installation" is perfectly fine with me! Please let me know if you want me to draft a PR, in case you'd prefer not to do it yourself.

**Reply by BurntSushi** [reply]
A PR would be great, thanks. :)

**Reply by nalgeon** [reply]
Ready for your review: #2760

**Comment by gcflymoto** [comment]
This is fantastic @nalgeon ! .. is it me or are the colors hard to read?

![image](https://github.com/BurntSushi/ripgrep/assets/252429/86e3a445-71f9-4bc4-90a0-468228e3c257)

**Reply by nalgeon** [reply]
Thank you! Yeah, the colors in dark mode could be better. I'm not very good with color palettes and contrast, to be honest 😅 But I'll look into it.
